api/models/monster.py

```diff
@@ -0,0 +1,120 @@
+
+import json
+
+from django.db import models
+
+from .models import GameContent
+from .spell import Spell
+
+
+class Monster(GameContent):
+    size = models.TextField(help_text='Monster size category.')
+    type = models.TextField(
+        help_text='The type of the monster, such as "aberration"')
+    subtype = models.TextField(
+        help_text='If applicable, the subtype of the monster, such as "shapechanger"')
+    group = models.TextField(
+        null=True,
+        help_text='Used to group similar creatures at different stages. "Green Dragon"')
+    alignment = models.TextField(
+        help_text='Short description of the creature alignment, such as "lawful good"')
+    armor_class = models.IntegerField(
+        default=12, help_text='Integer representing the armor class.')
+    armor_desc = models.TextField(
+        null=True, help_text='Description of the armor or armor type.')
+    hit_points = models.IntegerField(
+        null=True, help_text='Integer of the hit points.')
+    hit_dice = models.TextField(
+        help_text='Dice string representing a way to calculate hit points.')
+    speed_json = models.TextField()
+
+    def speed(self):
+        return json.loads(self.speed_json)
+    strength = models.IntegerField(
+        null=True, help_text='Integer representing the strength score.')
+    dexterity = models.IntegerField(
+        null=True, help_text='Integer represeting the dexterity score.')
+    constitution = models.IntegerField(
+        null=True, help_text='Integer representing the constitution score.')
+    intelligence = models.IntegerField(
+        null=True, help_text='Integer representing the intelligence score.')
+    wisdom = models.IntegerField(
+        null=True, help_text='Integer representing the wisdom score.')
+    charisma = models.IntegerField(
+        null=True, help_text='Integer representing the charisma score.')
+    strength_save = models.IntegerField(
+        null=True, help_text='Integer representing the strength save.')
+    dexterity_save = models.IntegerField(
+        null=True, help_text='Integer representing the dexterity save.')
+    constitution_save = models.IntegerField(
+        null=True, help_text='Integer representing the constitution save.')
+    intelligence_save = models.IntegerField(
+        null=True, help_text='Integer representing the intelligence save')
+    wisdom_save = models.IntegerField(
+        null=True, help_text='Integer representing the wisdom save.')
+    charisma_save = models.IntegerField(
+        null=True, help_text='Integer representing the charisma save.')
+    perception = models.IntegerField(
+        null=True, help_text='Integer representing the passive perception score.')
+    skills_json = models.TextField()
+
+    def skills(self):
+        return json.loads(self.skills_json)
+    damage_vulnerabilities = models.TextField(
+        help_text='Comma separated list of damage types the monster is vulnerable to.')
```

Comment:
Several of these fields are comma-separated lists. I'd like to explore representing the list more robustly as an actual list of values. You could imagine a scenario where we name a value with a comma in it (ex. "Paralyzed, A5E variant"), and then it's unclear how to parse. We wouldn't have that problem with a bona-fide list value.

I have two suggestions for how to implement this.

1. Use a JSONField to store a list of values. The downside is that the values in that list couldn't be Choices as I described above. We would need to store the string values directly.
2. Create a Model corresponding to each Enum: for example we'd define `class DamageType(models.TextChoices)` and also `class DamageTypeModel(models.Model)`. Then in the Monster model, define columns like `damage_vulnerabilities = models.ManyToManyField(DamageTypeModel)`. I think this is actually a very clean solution: the backend could be very simple, since the enum models never rely on anything else; and from a reader perspective, they would look just like a list of their stringy values. But, it is a little work to refactor.

(Third bad solution, migrate to Postgres, where we have an ArrayField. Lol.)

What do you think? If you think it sounds worthwhile but like a pain-in-the-butt to implement, I could find time for the refactor.